File: frontend/types/file.ts
export type TreeNode = {
    name: string;
    path: string;
    type: "folder" | "file";
    file_id?: number | null;
    children: TreeNode[];
  };

  export type FileDetail = {
    id: number;
    project_id: number;
    scan_run_id: number;
    path: string;
    filename: string;
    extension?: string | null;
    language?: string | null;
    size_bytes: number;
    content_preview?: string | null;
    summary?: string | null;
    is_supported: boolean;
    created_at: string;
  };

File: backend/app/schemas/file.py
from pydantic import BaseModel
from datetime import datetime

class FileResponse(BaseModel):
    id: int
    project_id: int
    scan_run_id: int
    path: str
    filename: str
    extension: str | None = None
    language: str | None = None
    size_bytes: int
    content_preview: str | None = None
    summary: str | None = None
    is_supported: bool
    created_at: datetime

    class Config:
        from_attributes = True

class TreeNode(BaseModel):
    name: str
    path: str
    type: str
    file_id: int | None = None
    children: list["TreeNode"] = []

TreeNode.model_rebuild()
